--- a/build/common.gypi	2016-07-04 11:59:50.871247305 +0200
+++ b/build/common.gypi	2016-07-04 12:15:01.996579428 +0200
@@ -6027,16 +6027,6 @@
         ['CXX.host', '<(host_cxx)'],
       ],
     }],
-    ['OS=="linux" and target_arch=="arm" and host_arch!="arm" and chromeos==0 and clang==0', {
-      # Set default ARM cross tools on linux.  These can be overridden
-      # using CC,CXX,CC.host and CXX.host environment variables.
-      'make_global_settings': [
-        ['CC', '<!(which arm-linux-gnueabihf-gcc)'],
-        ['CXX', '<!(which arm-linux-gnueabihf-g++)'],
-        ['CC.host', '<(host_cc)'],
-        ['CXX.host', '<(host_cxx)'],
-      ],
-    }],
     ['use_goma==1', {
       'make_global_settings': [
        ['CC_wrapper', '<(gomadir)/gomacc'],
