--- a/build/config/linux/pkg-config.py	2016-07-04 11:57:19.578378218 +0200
+++ b/build/config/linux/pkg-config.py	2016-07-13 18:49:35.842922330 +0200
@@ -208,6 +208,9 @@
     else:
       cflags.append(flag)
 
+  # remove empty values
+  includes = [x for x in includes if x]
+
   # Output a GN array, the first one is the cflags, the second are the libs. The
   # JSON formatter prints GN compatible lists when everything is a list of
   # strings.
